language/bert/colotensor/utils.py

```diff
@@ -0,0 +1,274 @@
+from tqdm import tqdm
+import time
+import math
+import os
+
+import colossalai.utils as utils
+import torch
+from torch.distributed import all_reduce, get_rank, get_world_size, is_initialized
+from colossalai.utils import colo_set_process_memory_fraction, get_current_device, MultiTimer
+from colossalai.core import global_context as gpc
+from concurrent.futures import ThreadPoolExecutor
+
+class AsyncMemoryMonitor:
```

Comment:
Why is this needed?

Reply:
Because engine is not used, I use it to show statistics.